src/Plugins/ComplexCore/test/RawBinaryReaderTest.cpp

```diff
@@ -0,0 +1,446 @@
+ *
+ *  Case3: This tests when the wrong component size is chosen. (The total number of scalar elements in the file does not evenly divide by the chosen component size).
+ *
+ *  Case4: This tests when skipHeaderBytes is non-zero, and checks to see if the data read is the same as the data written.
+ *
+ *  Case5: This tests when skipHeaderBytes equals the file size
+ */
+
+/** we are going to use a fairly large array size because we want to exercise the
+ * possibilty that we can not write the data or read the data from the file in one
+ * step in the filter
+ */
+
+#include <catch2/catch.hpp>
+
+#include "complex/Common/ScopeGuard.hpp"
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/DataStore.hpp"
+#include "complex/Parameters/ArrayCreationParameter.hpp"
+#include "complex/Parameters/ChoicesParameter.hpp"
+#include "complex/Parameters/FileSystemPathParameter.hpp"
+#include "complex/Parameters/NumberParameter.hpp"
+#include "complex/Parameters/NumericTypeParameter.hpp"
+
+#include <filesystem>
+namespace fs = std::filesystem;
+
+#include "ComplexCore/Filters/RawBinaryReaderFilter.hpp"
+#include "complex/unit_test/complex_test_dirs.hpp"
+
+using namespace complex;
+
+const std::string k_TestOutput = (fs::path(complex::unit_test::k_BinaryDir.str()).append("RawBinaryReaderTest").append("Output.bin")).string();
+const DataPath k_CreatedArrayPath = DataPath(std::vector<std::string>{"Test_Array"});
+
+constexpr int32_t k_RbrNumComponentsError = -392;
+constexpr int32_t k_RbrWrongType = -393;
+constexpr int32_t k_RbrSkippedTooMuch = -395;
+
+namespace Detail
+{
+enum Endian
+{
+  Little = 0,
+  Big
+};
+} // namespace Detail
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+Arguments createFilterArguments(complex::NumericType scalarType, usize N, usize skipBytes)
+{
+  Arguments args;
+
+  args.insertOrAssign(RawBinaryReaderFilter::k_InputFile_Key, std::make_any<FileSystemPathParameter::ValueType>(fs::path(k_TestOutput)));
+  args.insertOrAssign(RawBinaryReaderFilter::k_ScalarType_Key, std::make_any<NumericType>(scalarType));
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<uint64>(N));
+  args.insertOrAssign(RawBinaryReaderFilter::k_Endian_Key, std::make_any<ChoicesParameter::ValueType>(Detail::Little));
+  args.insertOrAssign(RawBinaryReaderFilter::k_SkipHeaderBytes_Key, std::make_any<uint64>(skipBytes));
+  args.insertOrAssign(RawBinaryReaderFilter::k_CreatedAttributeArrayPath_Key, k_CreatedArrayPath);
+
+  return args;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+template <typename T>
+bool createAndWriteToFile(std::string filePath, T* dataArray, usize dataSize)
```

Comment:
```suggestion
bool createAndWriteToFile(const std::string& filePath, nonstd::span<T> data)
```
`std::string` should be passed by reference, and the pointer and size can be replaced with `nonstd::span<T>`.